src/OrchardCore.Modules/OrchardCore.Contents/Workflows/Activities/ContentForEachTask.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.Extensions.Localization;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement.Workflows;
+
+namespace OrchardCore.Contents.Workflows.Activities
+{
+    public class ContentForEachTask : TaskActivity
+    {
+        readonly IStringLocalizer S;
+        private readonly ISession _session;
+        private readonly IContentManager _contentManager;
+
+        public ContentForEachTask(IContentManager contentManager, IStringLocalizer<RetrieveContentTask> localizer, ISession session)
+        {
+            S = localizer;
+            _session = session;
+            _contentManager = contentManager;
+        }
+        public override string Name => nameof(ContentForEachTask);
+
+        public override LocalizedString DisplayText => S["Content For Each Task"];
+
+        public override LocalizedString Category => S["Content"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Iterate"], S["Done"]);
+        }
+        public override bool CanExecute(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return true;
+        }
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var contentItems = await _session
+                .Query<ContentItem, ContentItemIndex>(index => index.ContentType == ContentType && index.Published == true && index.Latest == true)
```

Comment:
The filter doesn't seem right, Published AND Latest? Means if there is a draft we are not loading the Published version of it.

Reply:
My use case was for published only, I will add options to the view model.

Reply:
Still, if there is a published item and there is also a draft, you are not loading the published one.

Reply:
My misunderstanding of the filters, will sort.